graphium/trainer/predictor.py

```diff
@@ -608,7 +617,7 @@ def on_train_start(self):
 
     def get_progress_bar_dict(self) -> Dict[str, float]:
         prog_dict = {}
-        prog_dict["loss"] = self.task_epoch_summary.weighted_loss.item()
+        prog_dict["loss"] = self.task_epoch_summary.weighted_loss.detach().cpu()
```

Comment:
Not sure if it is related to the detach thing, I've tried to run this branch on IPU with the debug config, but it gives me strange compilation error. Is it possible to have .detach() (also other places) for CPU and GPU only?

Reply:
I see. It is supposed to be a performance optimization, following [https://pytorch-lightning.readthedocs.io/en/1.3.8/benchmarking/performance.html#item-numpy-cpu](https://pytorch-lightning.readthedocs.io/en/1.3.8/benchmarking/performance.html#item-numpy-cpu). However, I'm not sure it improves a lot. If it causes cross-hardware issues, I'd rather undo it for now. Thanks for pointing this out.

Reply:
@zhiyil-graphcore any update on that?

Reply:
The compilation looks ok with this branch now. The compilation error might come from something else before.